File: Array/BinarySearch/BinarySearch.js
function binarySearch(arr, target) {
  let start = 0;
  let end = arr.length - 1;
  let mid = ~~((start + end) / 2);
  while (start <= end) {
    if (arr[mid] === target) {
      return mid;
    }

    if (target > arr[mid]) start = mid + 1;

    if (target < arr[mid]) end = mid - 1;

    mid = ~~((start + end) / 2);
  }

  mid = -1;
  return mid;
}

let arr = [1, 2, 4, 12, 52, 59];
let target = 12;

console.log(binarySearch(arr, target));

File: Array/Recursion/binarySearch.py
def binarySearch(arr, start, end, target):
    if start > end:
        return False

    mid = (start + end) // 2
    if arr[mid] == target:
        return True

    if arr[mid] < target:
        return binarySearch(arr, start + 1, end, target)

    else:
        return binarySearch(arr, start, end - 1, target)


print(binarySearch([2, 3, 5, 6, 7], 0, 4, 8))
